inspirehep/testlib/api/holdingpen.py

```diff
@@ -89,6 +134,120 @@ def get_detail_entry(self, holdingpen_id):
+
+        Returns:
+            requests.Response: The actual http response to the last call (the
+                actual /resolve endpoint).
+
+        Raises:
+            requests.exceptions.BaseHttpError: any error related to the http
+                calls made.
+
+        Example:
+            >>> my_entry = holdingpen_client.get_detail_entry(holdingpen_id=1234)
+            >>> my_entry.core = False   # do some changes
+            >>> holdingpen_client._edit_workflow(holdingpen_entry=my_entry)
+            <Response [200]>
+
+        """
+
+        edit_response = self._client.post(
+            self.HOLDINGPEN_EDIT_URL.format(workflow_id=holdingpen_entry.workflow_id),
+            json=holdingpen_entry.to_json(),
+        )
+        edit_response.raise_for_status()
+        return edit_response
+
+    def _resolve_workflow(self, holdingpen_entry, resolution_data):
+        """
+        Helper method to resolve a workflow action, for more details see:
+            https://github.com/inveniosoftware-contrib/invenio-workflows-ui/blob/master/invenio_workflows_ui/api.py#L272
+
+        Args:
+            holdingpen_id(int): id of the holdingpen entry to accept.
+            resolution_data(dict): Data to pass to the resolve endpoint.
+
+        Returns:
+            requests.Response: The actual http response to the last call (the
+                actual /resolve endpoint).
+
+        Raises:
+            requests.exceptions.BaseHttpError: any error related to the http
+                calls made.
+
+        Example:
+            >>> holdingpen_client._resolve_workflow(
+                holdingpen_id=12345,
+                resolution_data={
+                    'pdf_upload': false,
+                    'reason': '',
+                    'value': 'accept_core',
+                }
+            )
+            <Response [200]>
+        """
+        resolve_response = self._client.post(
+            self.HOLDINGPEN_RESOLVE_URL.format(workflow_id=holdingpen_entry.workflow_id),
+            json=resolution_data,
+        )
+        resolve_response.raise_for_status()
+        return resolve_response
+
+    def _resolve_hep_approval(self, holdingpen_id, approval_type, reason=''):
+        """
+        Args:
+            holdingpen_id(int): id of the holdingpen entry to accept.
+            approval_type(str): Approval resolution, currently one of
+                'accept_core', 'accept_non_core' or 'reject'
+            reason(str): If it was rejected, the reason explaining why.
+            pdf_upload(bool): If a new pdf was uploadad
+
+        Returns:
+            requests.Response: the response to the last 'resolve' api call.
+        """
+        resolution_data = {
+            'pdf_upload': False,
+            'reason': reason,
+            'value': approval_type,
+        }
+
+        entry_to_accept = self.get_detail_entry(holdingpen_id=holdingpen_id)
+        self._edit_workflow(holdingpen_entry=entry_to_accept)
```

Comment:
We're saving the same workflow we just retrieved: Antonio explained that it's just something that happens, and we don't really know why, but maybe worth to add a comment explaining that?

Reply:
The only reason is to make sure that any change that was made in the non-editor ui (the regular details page, you can edit stuff there too). It could actually be removed from here, but I'd prefer leaving it until we remove it from the ui too, to do the same calls it does.